src/cljam/util/intervals.clj

```diff
@@ -17,12 +17,50 @@
        (into (sorted-map))
        (->SortedMapIntervals)))
 
+(defn- find-nclist-overlap-intervals [nclist start end]
+  (when-let [target-intervals
+             (->> (subseq nclist  >= start)
+                  (map second)
+                  (take-while #(<= (:start (first %)) end)))]
+    (mapcat #(cons (first %)
+                   (find-nclist-overlap-intervals (second %)
+                                                  start end))
+            target-intervals)))
```

Comment:
Although it wouldn't cause a real problem, the conditional of this `when-let` will always be evaluated to a truthy value since `take-while` always returns a non-nil value. You can rewrite this as:

```clojure
(some->> (subseq nclist >= start)
         (map second)
         (take-while #(<= (:start (first %)) end))
         seq
         (mapcat #(cons (first %)
                        (find-nclist-overlap-intervals (second %)
                                                       start end))))
```

Or simply:

```clojure
(->> (subseq nclist >= start)
     (map second)
     (take-while #(<= (:start (first %)) end))
     (mapcat #(cons (first %)
                    (find-nclist-overlap-intervals (second %)
                                                   start end))))
```